phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -3243,10 +3243,10 @@ public boolean isViewReferenced() {
                 tableUpsert.setInt(34, ttl);
             }
 
-            if (rowKeyPrefix == null) {
+            if (rowKeyMatcher == null) {
```

Comment:
@jpisaac a question do we want to store null in SYSCAT in case of Empty_Byte_array as well!?
We are not doing any scans based on it bring null!?

Reply:
Added that check too!